### PR TITLE
Support for Distribution path mapping with AQL

### PR DESCRIPTION
**Pull request by YardenEdery**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

**Comment by YardenEdery** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by YardenEdery** [comment]
Not sure about adding tests here, there are none related to distribution

File: common/spec/specfiles.go [hunk 1]
```diff
@@ -194,6 +197,9 @@ func ValidateSpec(files []File, isTargetMandatory, isSearchBasedSpec bool) error
 		isBypassArchiveInspection, _ := file.IsBypassArchiveInspection(false)
 		isTransitive, _ := file.IsTransitive(false)
 
+		if isPathMapping && !isAql {
+			return errorutils.CheckErrorf("mappings supported only with Aql")
```
Comment (eyalbe4) [inline common/spec/specfiles.go:201]: mappings supported only with Aql
-->
path mapping is supported only with Aql

File: common/spec/specfiles.go [hunk 2]
```diff
@@ -206,6 +212,12 @@ func ValidateSpec(files []File, isTargetMandatory, isSearchBasedSpec bool) error
 		if isSearchBasedSpec && !isAql && !isPattern && !isBuild && !isBundle {
 			return errorutils.CheckErrorf("spec must include either aql, pattern, build or bundle")
 		}
+		if isPathMapping && isTarget {
+			return fileSpecValidationError("mappings", "target")
+		}
+		if isPathMapping && isPattern {
+			return fileSpecValidationError("mappings", "pattern")
+		}
```
Comment (eyalbe4) [inline common/spec/specfiles.go:220]: Let's place the above line right after the following code for better readability - 
```
if isPathMapping && !isAql {
	return errorutils.CheckErrorf("mappings supported only with Aql")
}
```

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
Let's add a complete description for this PR.

File: common/spec/specfiles.go (hunk 1 above)
Comment (YardenEdery) [reply common/spec/specfiles.go:201]: Are you sure because "mappings" is the interface to the user, they will use "mappings" in their spec.json and I saw in the code you treat it as PathMappings so I kept the same conventions, and thought that the error should show the user what he needs to fix.

Comment (eyalbe4) [reply common/spec/specfiles.go:201]: OK. NP.

File: common/spec/specfiles.go [hunk 3]
```diff
@@ -41,6 +41,7 @@ func CreateSpecFromFile(specFilePath string, specVars map[string]string) (spec *
 
 type File struct {
 	Aql                     utils.Aql
+	PathMapping             utils.PathMapping `json:"mappings,omitempty"`
```
Comment (yahavi) [inline common/spec/specfiles.go:44]: Can we leave it "pathMapping"? Or use "mapping"?
```suggestion
	PathMapping             utils.PathMapping
```

**Review by yahavi — CHANGES_REQUESTED** [review]

File: common/spec/specfiles.go (hunk 3 above)
Comment (YardenEdery) [reply common/spec/specfiles.go]: I tried to keep the convention, I saw in lifecycle directory the use of pattern/target (which later on transform to "mappings") as "PathMappings", thus tried to keep it the same. I can change it to `Mappings` in the `File struct` but eventually its similar with lifecycles "PathMapping" which I preferred not to touch / delete / modify.
WDYT?
youll see what im talking about by searching in the code (jfrog-client-go):
```
 type Modifications struct {
	PathMappings []utils.PathMapping `json:"mappings"`
}

type PathMapping struct {
	Pattern string
	Target  string
}
```

I do think that changing to "mappings" will be better because thats how distribution server expects it eventually

**Review by yahavi — DISMISSED** [review]
LGTM
